src/main/java/br/com/eventvs/domain/controller/InscricaoController.java

```diff
@@ -20,13 +24,48 @@
 	@Autowired
 	private InscricaoRepository inscricaoRepository;
 	
+	@Autowired
+	private EventoRepository eventoRepository;
+	
 	@Autowired
 	private ParticipanteRepository participanteRepository;
 	
 	@Autowired
 	private PessoaRepository pessoaRepository;
 	
+	/**
+     * Cadastra uma inscricao
+     *
+     * @param InscricaoRequest
+     * @return Inscricao
+     * @return null
+     * */
+	public Inscricao cadastrarInscricao(InscricaoRequest inscricaoRequest) {
+		Optional <Participante> participante = participanteRepository.findById(inscricaoRequest.getParticipante_id());
+		Optional <Evento> evento = eventoRepository.findById(inscricaoRequest.getEvento_id());
+		if(participante.isPresent() && evento.isPresent()) {
+			//Checagem pra saber se a inscricao já existe
+			List<Inscricao> inscricoes = inscricaoRepository.findByParticipante(participante.get());
+			for(int i =0; i< inscricoes.size();i++) {
+				if(inscricoes.get(i).getEvento().equals(evento.get())) {
+					return null;
+				}
+			}
```

Comment:
Maluco tais me tirando? kkkkkkk
faz uma consulta melhor pra n precisar desse for
algo como findByParticipanteAndEvento

Reply:
Vai se fuder irmão. vc quer o codigo inteiro sem 1 for é

Reply:
claro fiao, se vc pode fazer uma parada mais safe, faz essa parada